be/src/olap/txn_manager.cpp

```diff
@@ -126,9 +126,8 @@ OLAPStatus TxnManager::add_txn(
     return OLAP_SUCCESS;
 }
 
-OLAPStatus TxnManager::delete_txn(
-    TPartitionId partition_id, TTransactionId transaction_id,
-    TTabletId tablet_id, SchemaHash schema_hash) {
+OLAPStatus TxnManager::delete_txn(TPartitionId partition_id, TTransactionId transaction_id,
+                                  TTabletId tablet_id, SchemaHash schema_hash) {
 
     pair<int64_t, int64_t> key(partition_id, transaction_id);
     TabletInfo tablet_info(tablet_id, schema_hash);
```

Comment:
delete txn should not do clear pending data task, because a txn is published and will call delete_txn, if clear pending rowset, then the visible data is deleted.
the caller method should call rowset. remove to delete the data. or add the delete rowset to a queue and write a single gc thread to gc it.